File: demo/app/src/main/java/com/cyjh/mobileanjian/screencap/ForScreenShotActivity.java
package com.cyjh.mobileanjian.screencap;

import android.app.Activity;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.media.projection.MediaProjectionManager;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;

/* JADX INFO: loaded from: classes2.dex */
public class ForScreenShotActivity extends Activity {
    private static final String OoooOoO = "ForScreenShotActivity";
    public static final int OoooOoo = 32896;

    private void OooO00o() {
        if (Build.VERSION.SDK_INT >= 21) {
            startActivityForResult(((MediaProjectionManager) getSystemService("media_projection")).createScreenCaptureIntent(), OoooOoo);
        } else {
            Log.e(OoooOoO, "The API version is too low,required is >= 21.");
            finish();
        }
    }

    @Override // android.app.Activity
    public void onActivityResult(int i, int i2, Intent intent) {
        super.onActivityResult(i, i2, intent);
        if (i != 32896) {
            return;
        }
        if (i2 != -1 || intent == null) {
            OooO00o();
            setResult(0);
        } else {
            setResult(-1);
            ScreenShoterV3.getInstance().init(this, intent);
            finish();
        }
    }

    @Override // android.app.Activity
    public void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        requestWindowFeature(1);
        getWindow().setBackgroundDrawable(new ColorDrawable(0));
        getWindow().setDimAmount(0.0f);
        OooO00o();
    }
}

File: app/src/main/java/com/example/myapplication/screencap/ForScreenShotActivity.kt
package com.example.myapplication.screencap

import android.app.Activity
import android.content.Intent
import android.graphics.drawable.ColorDrawable
import android.media.projection.MediaProjectionManager
import android.os.Build
import android.os.Bundle
import android.util.Log

/**
 * MediaProjection 权限请求 Activity - 参考 ForScreenShotActivity.java
 * 透明 Activity，仅用于请求截屏权限
 */
class ForScreenShotActivity : Activity() {

    companion object {
        private const val TAG = "ForScreenShotActivity"
        const val REQUEST_CODE = 32896
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(android.view.Window.FEATURE_NO_TITLE)
        window.setBackgroundDrawable(ColorDrawable(0))
        window.setDimAmount(0f)
        requestScreenCapture()
    }

    private fun requestScreenCapture() {
        if (Build.VERSION.SDK_INT >= 21) {
            val mgr = getSystemService(MEDIA_PROJECTION_SERVICE) as MediaProjectionManager
            startActivityForResult(mgr.createScreenCaptureIntent(), REQUEST_CODE)
        } else {
            Log.e(TAG, "API version too low, requires >= 21")
            finish()
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode != REQUEST_CODE) return

        if (resultCode != RESULT_OK || data == null) {
            // 用户拒绝，重新请求
            requestScreenCapture()
            setResult(RESULT_CANCELED)
        } else {
            setResult(RESULT_OK)
            ScreenShoterV3.getInstance().init(this, data)
            finish()
        }
    }
}
